docs/guide/content/development_guide/e2e_tests.md

```diff
@@ -0,0 +1,226 @@
+
+### Test Structure
+
+```
+e2e_tests/
+├── cypress/
+│   ├── e2e/                    # Test specifications
+│   ├── plugins/                # Cypress plugins and utilities
+│   │   ├── config.js          # cy.loop configuration object
+│   │   ├── loop.js            # Main plugin configuration
+│   │   └── navigation.js      # Navigation utilities
+│   ├── support/               # Cypress support files
+│   │   └── e2e.js            # Global test setup
+│   └── fixtures/              # Test data files
+├── cypress.config.js          # Cypress configuration
+├── package.json              # Node.js dependencies
+├── Makefile                  # Test automation commands
+└── docker-compose.yml        # Test environment setup
+```
+
+## Running Tests Locally
+
+### Prerequisites
+
+- Docker and Docker Compose installed
+- Make utility
+- Access to test credentials (see [Local Environment](local_environment.md))
+
+### Quick Start
+
+1. **Start the test environment:**
+   ```bash
+   cd e2e_tests
+   make env_up
+   ```
+   This command starts OnDemand Loop and all required services using Docker Compose.
+
+2. **Build Cypress dependencies:**
+   ```bash
+   make cypress_build
+   ```
+
+3. **Run tests:**
+   ```bash
+   # Run all tests headless
+   make cypress_run
+   
+   # Run tests interactively (requires X11 forwarding)
+   make cypress_open
+   ```
+
+4. **Clean up:**
+   ```bash
+   make clean
+   ```
+
+### Available Make Targets
+
+| Target | Description |
+|--------|-------------|
+| `env_up` | Start test environment (OOD, mocks, services) |
+| `env_down` | Stop test environment |
+| `env_status` | Show status of environment services |
+| `env_logs` | Show logs from environment services |
+| `cypress_deps` | Generate/update package-lock.json using Docker container |
+| `cypress_build` | Build Cypress project and install dependencies |
+| `cypress_run` | Run Cypress tests headless |
+| `cypress_open` | Open Cypress interactive test runner |
+| `clean` | Stop environment and clean up artifacts |
+
+### Environment Variables
+
+Configure tests using these environment variables:
+
+| Variable | Description | Default |
+|----------|-------------|---------|
+| `CYPRESS_SPEC` | Specific test spec to run | All tests |
+| `LOOP_USERNAME` | Username for OOD authentication | - |
+| `LOOP_PASSWORD` | Password for OOD authentication | - |
+| `OOD_VERSION` | OnDemand version | `3.1.7` |
+
+Example:
+```bash
+CYPRESS_SPEC=cypress/e2e/homepage.cy.js make cypress_run
+```
+
+## Writing Tests
+
+### Test Structure
+
+Follow this pattern for new tests:
+
+```javascript
+import { visitLoopRoot } from '../plugins/navigation'
+
+describe('Feature Name', () => {
+  it('should perform expected behavior', () => {
+    // Navigate to the application
+    visitLoopRoot()
+    
+    // Interact with elements
+    cy.get('[data-cy=element]').click()
+    
+    // Assert expected results
+    cy.contains('Expected Text').should('be.visible')
+  })
+})
+```
```

Comment:
I would double-check code formatters for the markdown specific for the documentation engine. It seems to be different than the Github one

Reply:
fixed, missing code highlight config